Source/WebKit/Platform/IPC/StreamServerConnection.h

```diff
@@ -154,18 +163,20 @@ void StreamServerConnection::sendAsyncReply(AsyncReplyID asyncReplyID, Arguments
     m_connection->sendSyncReply(WTFMove(encoder));
 }
 
-inline void StreamServerConnection::Handle::encode(Encoder& encoder) const
+inline void StreamServerConnection::Handle::encode(Encoder& encoder)
 {
-    encoder << outOfStreamConnection << buffer;
+    encoder << WTFMove(outOfStreamConnection) << buffer;
 }
 
 inline std::optional<StreamServerConnection::Handle> StreamServerConnection::Handle::decode(Decoder& decoder)
 {
+    std::optional<Handle> result;
     auto outOfStreamConnection = decoder.decode<Connection::Handle>();
     auto buffer = decoder.decode<StreamConnectionBuffer::Handle>();
     if (UNLIKELY(!decoder.isValid()))
-        return std::nullopt;
-    return Handle { WTFMove(*outOfStreamConnection), WTFMove(*buffer) };
+        return result;
+    result.emplace(Handle(WTFMove(*outOfStreamConnection), WTFMove(*buffer)));
```

Comment:
I think if you instead `return std::optional<Handle> { WTFMove(*outOfStreamConnection), WTFMove(*buffer) };` you get the inplace constructor that works with move-only types?